src/librustdoc/passes/collect_trait_impls.rs

```diff
@@ -62,9 +60,7 @@ crate fn collect_trait_impls(krate: Crate, cx: &mut DocContext<'_>) -> Crate {
     ) {
         if let Some(target) = map.get(&type_did) {
             debug!("add_deref_target: type {:?}, target {:?}", type_did, target);
-            if let Some(target_prim) = target.primitive_type() {
-                cleaner.prims.insert(target_prim);
-            } else if let Some(target_did) = target.def_id_no_primitives() {
+            if let Some(target_did) = target.def_id_no_primitives() {
                 // `impl Deref<Target = S> for S`
```

Comment:
This in particular makes me worried we'll no longer show Deref methods if an item derefs to a primitive type. Can you add a test for that case?